Core/Core/Assignments/AssignmentList/ViewModel/AssignmentListViewModel.swift

```diff
@@ -128,41 +143,31 @@ public class AssignmentListViewModel: ObservableObject {
         || ![0, AssignmentFilterOption.allCases.count].contains(selectedFilterOptions.count)
             && selectedFilterOptions != initialFilterOptions
 
-        assignmentGroups = env.subscribe(GetAssignmentsByGroup(courseID: courseID, gradingPeriodID: gradingPeriod?.id)) { [weak self] in
-            self?.assignmentGroupsDidUpdate()
-        }
-        assignmentGroups.refresh()
-    }
-
-    public func viewDidAppear() {
-        loadAssignmentListPreferences()
-        gradingPeriods.refresh()
-        filterOptionsDidUpdate(filterOptions: selectedFilterOptions, gradingPeriod: defaultGradingPeriod)
-        course.refresh()
-        assignmentGroups.refresh(force: true)
-
-        isFilterIconSolid = ![0, AssignmentFilterOption.allCases.count].contains(selectedFilterOptions.count)
+        assignmentGroupsDidUpdate()
     }
 
     private func assignmentGroupsDidUpdate() {
         if !assignmentGroups.requested || assignmentGroups.pending { return }
+        if !gradingPeriods.requested || gradingPeriods.pending { return }
 
         isShowingGradingPeriods = gradingPeriods.count > 1
         var assignmentGroupViewModels: [AssignmentGroupViewModel] = []
-        let assignments: [Assignment] = filterAssignments(assignmentGroups.compactMap { $0 })
+        let compactAssignmentGroups = assignmentGroups
+            .compactMap { $0.assignments }
+            .map { Array($0) }
+            .flatMap { $0 }
+        let assignments: [Assignment] = filterAssignments(compactAssignmentGroups)
 
         switch selectedSortingOption {
         case .groupName:
-            for section in 0..<(assignmentGroups.sections?.count ?? 0) {
-                if let group = assignmentGroups[IndexPath(row: 0, section: section)]?.assignmentGroup {
-                    let groupAssignments: [Assignment] = assignments.filter { $0.assignmentGroup == group }
-                    if !groupAssignments.isEmpty {
-                        assignmentGroupViewModels.append(AssignmentGroupViewModel(
-                            assignmentGroup: group,
-                            assignments: groupAssignments,
-                            courseColor: courseColor
-                        ))
-                    }
+            assignmentGroups.forEach { group in
+                let groupAssignments: [Assignment] = assignments.filter { $0.assignmentGroup == group }
+                if !groupAssignments.isEmpty {
```

Comment:
```suggestion
                guard groupAssignments.isNotEmpty else { return }
```

Reply:
There's no such thing as`groupAssignments.isNotEmpty` so I kept `!groupAssignments.isEmpty` but changed the `if` to `guard`.

Reply:
Huh, I see now that `isNotEmpty` is only added for strings at the moment, my bad :)